### PR TITLE
Renamed visit_expr() to expr_leaf_spans() in sfv2

### PR DESCRIPTION
**Pull request by Tusharkshahi**
- Renamed visit_expr() to expr_leaf_spans() in sfv2 #2509

- Used while let in place of for in peekable() iterations in sfv2 #2496

### CODE REVIEW

**Comment by eureka-cpu** [comment]
@Tusharkshahi thanks for the contribution! There are a few things failing in CI, I'll go ahead and convert this PR to a draft for now and when you're ready for review just re-open it as ready (:

File: sway-fmt-v2/src/items/item_enum.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ impl Format for ItemEnum {
                 });
 
                 let mut value_pairs_iter = value_pairs.iter().enumerate().peekable();
-                for (var_index, (type_field, comma_token)) in value_pairs_iter.clone() {
+                while let some((var_index, (type_field, comma_token))) = value_pairs_iter.next() {
```
Comment (kayagokalp) [inline sway-fmt-v2/src/items/item_enum.rs:66]:
```suggestion
                while let Some((var_index, (type_field, comma_token))) = value_pairs_iter.next() {
```

I think this is why `clippy` and other CI steps are failing. If you could change occurrences of `some(..)` to `Some(..)`, it should be good to go!

**Review by kayagokalp — COMMENTED** [review]
Thanks for the contribution! Looks like you got lowercase `s` in your usage of `Some`. If you could convert occurrences of `some` to `Some` in your changes, this should be good to go!

Also I think you need to format the code, you can do that by running `cargo fmt` 👍

**Comment by sezna** [comment]
It looks like CI is failing in three places:
1. [the format check](https://github.com/FuelLabs/sway/runs/7792672977?check_suite_focus=true#step:4:12), which can be fixed by running `cargo fmt` in your code.
2. [clippy](https://github.com/FuelLabs/sway/runs/7792672750?check_suite_focus=true#step:5:774), which I'm actually not sure why is failing on this branch. Your changes appear to be unrelated. I'll look into that.
3. [the correctness checks for the formatter](https://github.com/FuelLabs/sway/runs/7792673474?check_suite_focus=true#step:5:990). Some of the changes in this PR are altering the functionality of the formatter and causing it to behave differently. You can run these tests locally by running `cargo test` in the formatter v2 directory and debug accordingly 😄 hint: cloning an iterator allows you to consume the clone without progressing the original iterator that was cloned.

**Comment by kayagokalp** [comment]
As #2518 merged, unrelated CI failure should be resolved. I merged master here to resolve that here

**Comment by eureka-cpu** [comment]
@Tusharkshahi let us know if there's anything we can help with!

**Comment by mitchmindtree** [comment]
@Tusharkshahi I'm going to close this PR as it's gathering dust, however if you'd like to rebase and take on the suggestions above feel free to re-open!

**Pull request closed without merging**